Settings.hs

```diff
@@ -78,6 +79,16 @@ instance FromJSON AppSettings where
         appCopyright              <- o .: "copyright"
         appAnalytics              <- o .:? "analytics"
 
+        -- This code enables MySQL's strict mode, without which MySQL will truncate data.
+        -- See https://github.com/yesodweb/persistent/wiki/Database-Configuration#strict-mode for details
+        -- If you choose to keep strict mode enabled, it's recommended that you enable it in your my.cnf file so that it's also enabled for your MySQL console sessions.
+        -- (If you enable it in your my.cnf file, you can delete this code).
+        let appDatabaseConf = fromYamlAppDatabaseConf { myConnInfo = (myConnInfo fromYamlAppDatabaseConf) {
+                MySQL.connectOptions =
+                  ( MySQL.connectOptions (myConnInfo fromYamlAppDatabaseConf)) ++ [MySQL.InitCommand "SET SESSION sql_mode = 'STRICT_ALL_TABLES';\0"]
```

Comment:
The ByteString here is explicitly null-terminated to work around [this issue](https://github.com/bos/mysql/issues/16) (related PR: https://github.com/bos/mysql/pull/17).